packages/flutter_tools/test/general.shard/pub_dependencies_project_validator_test.dart

```diff
@@ -0,0 +1,77 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter_tools/src/base/file_system.dart';
+import 'package:flutter_tools/src/globals.dart' as globals;
+import 'package:flutter_tools/src/project.dart';
+import 'package:flutter_tools/src/project_validator.dart';
+import 'package:flutter_tools/src/project_validator_result.dart';
+
+import '../src/common.dart';
+import '../src/context.dart';
+
+void main() {
+  late FileSystem fileSystem;
+
+  group('PubDependenciesProjectValidator', () {
+
+    setUp(() {
+      fileSystem = globals.localFileSystem;
+    });
+
+    testWithoutContext('success ', () async {
+      final ProcessManager processManager = FakeProcessManager.list(<FakeCommand>[
+        const FakeCommand(
+          command: <String>['dart', 'pub', 'deps', '--json'],
+          stdout: '{"packages": [{"dependencies": ["abc"], "source": "hosted"}]}',
+        ),
+      ]);
+      final PubDependenciesProjectValidator validator = PubDependenciesProjectValidator(processManager);
+
+      final List<ProjectValidatorResult> result = await validator.start(
+          FlutterProject.fromDirectoryTest(fileSystem.currentDirectory)
+      );
+      const String expected = 'All dependencies are hosted';
+      expect(result.length, 1);
+      expect(result[0].value, expected);
+      expect(result[0].status, StatusProjectValidator.success);
+    });
+
+    testWithoutContext('error ', () async {
+      final ProcessManager processManager = FakeProcessManager.list(<FakeCommand>[
+        const FakeCommand(
+          command: <String>['dart', 'pub', 'deps', '--json'],
+          stdout: 'stdout command fail',
```

Comment:
what does this mean?

Reply:
Just expecting something not a json (eg: if you haven't updated your pub dependencies and the pubspec has changed pub dart deps returns a message to update them. I don't have the exact message so I just fake this)

Reply:
I could be wrong, but I would imagine if you pass `--json`, only valid JSON should be output to stdout, and any diagnostics would go to stderr.